### ISSUE TITLE
panic: runtime error: index out of range

### ISSUE THREAD

**Issue by urban-1** [victorcoder/dkron#236]
Hi all,

#### Summary

I get "panic: runtime error: index out of range" doing the following steps:

1. Start a new dkron container
2. Create a new job via the API (every 5 mins ls -ali /tmp)
3. Delete the job from the UI
4. Re-create the job (same specs and name)

#### Setup

- 2x zookeeper: version 3.4.9, debian 8 based container, jre 8
- 1x dkron: version 0.9.2, alpine based (actually the official one with changed entrypoint)

##### Step 1 and 2

Job added with:

```
$ curl -X POST -d '{"name": "test2", "schedule": "@every 5s", "command": "ls -ali /tmp", "role": "web:1"}' 172.17.0.4:8080/v1/jobs -vvv
* Hostname was NOT found in DNS cache
*   Trying 172.17.0.4...
* Connected to 172.17.0.4 (172.17.0.4) port 8080 (#0)
> POST /v1/jobs HTTP/1.1
> User-Agent: curl/7.35.0
> Host: 172.17.0.4:8080
> Accept: */*
> Content-Length: 86
> Content-Type: application/x-www-form-urlencoded
> 
* upload completely sent off: 86 out of 86 bytes
< HTTP/1.1 201 Created
< Content-Type: application/json; charset=UTF-8
< Location: /v1/jobs/test2
< X-Whom: 504404e2ade1
< Date: Sun, 29 Jan 2017 23:15:06 GMT
< Content-Length: 334
< 
{"name":"test2","schedule":"@every 5s","shell":false,"command":"ls -ali /tmp","owner":"","owner_email":"","success_count":0,"error_count":0,"last_success":"0001-01-01T00:00:00Z","last_error":"0001-01-01T00:00:00Z","disabled":false,"tags":null,"retries":0,"dependent_jobs":null,"parent_job":"","processors":null,"concurrency":"allow"}
* Connection #0 to host 172.17.0.4 left intact
```

dkron log:

```
/opt/local/dkron/dkron agent -server -backend zk -backend-machine 172.17.0.3:2181 -log-level=debug
2017/01/29 23:14:43 [DEBUG] plugin: starting plugin: /opt/local/dkron/dkron-processor-files []string{"/opt/local/dkron/dkron-processor-files"}
2017/01/29 23:14:43 [DEBUG] plugin: waiting for RPC address for: /opt/local/dkron/dkron-processor-files
2017/01/29 23:14:43 [DEBUG] plugin: dkron-processor-files: 2017/01/29 23:14:43 [DEBUG] plugin: plugin address: unix /tmp/plugin894818988
2017/01/29 23:14:43 [DEBUG] plugin: starting plugin: /opt/local/dkron/dkron-processor-log []string{"/opt/local/dkron/dkron-processor-log"}
2017/01/29 23:14:43 [DEBUG] plugin: waiting for RPC address for: /opt/local/dkron/dkron-processor-log
2017/01/29 23:14:43 [DEBUG] plugin: dkron-processor-log: 2017/01/29 23:14:43 [DEBUG] plugin: plugin address: unix /tmp/plugin771234044
2017/01/29 23:14:43 [DEBUG] plugin: starting plugin: /opt/local/dkron/dkron-processor-syslog []string{"/opt/local/dkron/dkron-processor-syslog"}
2017/01/29 23:14:43 [DEBUG] plugin: waiting for RPC address for: /opt/local/dkron/dkron-processor-syslog
2017/01/29 23:14:43 [DEBUG] plugin: dkron-processor-syslog: 2017/01/29 23:14:43 [DEBUG] plugin: plugin address: unix /tmp/plugin179447562
Starting Dkron agent...
time="2017-01-29T23:14:43Z" level=info msg="agent: Dkron agent starting" node=504404e2ade1 
time="2017-01-29T23:14:43Z" level=info msg="agent: joining: [] replay: true" node=504404e2ade1 
time="2017-01-29T23:14:43Z" level=debug msg="store: Backend config" backend=zk keyspace=dkron machines=[172.17.0.3:2181] node=504404e2ade1 
time="2017-01-29T23:14:43Z" level=info msg="api: Running HTTP server" address=":8080" node=504404e2ade1 
time="2017-01-29T23:14:43Z" level=debug msg="rpc: Registering RPC server" node=504404e2ade1 rpc_addr="172.17.0.4:6868" 
time="2017-01-29T23:14:43Z" level=info msg="agent: Running for election" node=504404e2ade1 
time="2017-01-29T23:14:43Z" level=info msg="agent: Listen for events" node=504404e2ade1 
time="2017-01-29T23:14:43Z" level=info msg="agent: Cluster leadership lost" node=504404e2ade1 
time="2017-01-29T23:14:43Z" level=info msg="agent: Cluster leadership acquired" node=504404e2ade1 
time="2017-01-29T23:14:43Z" level=debug msg="agent: Restarting scheduler" node=504404e2ade1 
time="2017-01-29T23:14:44Z" level=debug msg="agent: Received event" event=member-join node=504404e2ade1 
time="2017-01-29T23:14:44Z" level=debug msg="agent: Member event" event=member-join member=504404e2ade1 node=504404e2ade1 
time="2017-01-29T23:15:03Z" level=debug msg="store: Setting job" job=test2 json="{\"name\":\"test2\",\"schedule\":\"@every 5s\",\"shell\":false,\"command\":\"ls -ali /tmp\",\"owner\":\"\",\"owner_email\":\"\",\"success_count\":0,\"error_count\":0,\"last_success\":\"0001-01-01T00:00:00Z\",\"last_error\":\"0001-01-01T00:00:00Z\",\"disabled\":false,\"tags\":null,\"retries\":0,\"dependent_jobs\":null,\"parent_job\":\"\",\"processors\":null,\"concurrency\":\"allow\"}" node=504404e2ade1 
time="2017-01-29T23:15:03Z" level=debug msg="store: Retrieved leader from datastore" node=504404e2ade1 
time="2017-01-29T23:15:03Z" level=debug msg="agent: Received event" event="query: scheduler:restart" node=504404e2ade1 
time="2017-01-29T23:15:03Z" level=debug msg="agent: Restarting scheduler" node=504404e2ade1 
time="2017-01-29T23:15:03Z" level=debug msg="agent: Received ack" from=504404e2ade1 node=504404e2ade1 
time="2017-01-29T23:15:03Z" level=debug msg="scheduler: Stopping scheduler" node=504404e2ade1 
time="2017-01-29T23:15:03Z" level=debug msg="scheduler: Adding job to cron" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:06Z" level=debug msg="agent: Done receiving acks and responses" node=504404e2ade1 query="scheduler:restart" 
time="2017-01-29T23:15:08Z" level=debug msg="scheduler: Run job" job=test2 node=504404e2ade1 schedule="@every 5s" 
time="2017-01-29T23:15:08Z" level=debug msg="store: Retrieved job from datastore" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:08Z" level=debug msg="agent: Filtered nodes to run: []" node=504404e2ade1 
time="2017-01-29T23:15:08Z" level=debug msg="agent: Filtered tags to run: map[]" node=504404e2ade1 
time="2017-01-29T23:15:08Z" level=debug msg="agent: Sending query" job_name=test2 json="{\"execution\":{\"job_name\":\"test2\",\"started_at\":\"0001-01-01T00:00:00Z\",\"finished_at\":\"0001-01-01T00:00:00Z\",\"group\":1485731708001177710,\"attempt\":1},\"rpc_addr\":\"172.17.0.4:6868\"}" node=504404e2ade1 query="run:job" 
time="2017-01-29T23:15:08Z" level=debug msg="agent: Received event" event="query: run:job" node=504404e2ade1 
time="2017-01-29T23:15:08Z" level=debug msg="agent: Running job" at=2 node=504404e2ade1 payload="{\"execution\":{\"job_name\":\"test2\",\"started_at\":\"0001-01-01T00:00:00Z\",\"finished_at\":\"0001-01-01T00:00:00Z\",\"group\":1485731708001177710,\"attempt\":1},\"rpc_addr\":\"172.17.0.4:6868\"}" query="run:job" 
time="2017-01-29T23:15:08Z" level=info msg="agent: Starting job" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:08Z" level=debug msg="agent: Received ack" from=504404e2ade1 node=504404e2ade1 query="run:job" 
time="2017-01-29T23:15:08Z" level=debug msg="rpc: Received GetJob" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:08Z" level=debug msg="store: Retrieved job from datastore" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:08Z" level=debug msg="agent: GetJob by RPC" command="ls -ali /tmp" node=504404e2ade1 
time="2017-01-29T23:15:08Z" level=debug msg="agent: Received response" from=504404e2ade1 node=504404e2ade1 query="run:job" response="{\"job_name\":\"test2\",\"started_at\":\"2017-01-29T23:15:08.006166623Z\",\"finished_at\":\"0001-01-01T00:00:00Z\",\"node_name\":\"504404e2ade1\",\"group\":1485731708001177710,\"attempt\":1}" 
time="2017-01-29T23:15:08Z" level=debug msg="store: Setting key" execution=1485731708006166623-504404e2ade1 job=test2 node=504404e2ade1 
time="2017-01-29T23:15:08Z" level=debug msg="proc: Command output" node=504404e2ade1 output=total 8
     46 drwxrwxrwt    2 root     root          4096 Jan 29 23:14 .
      2 drwxr-xr-x   32 root     root          4096 Jan 29 23:14 ..
     49 srwxr-xr-x    1 root     root             0 Jan 29 23:14 plugin179447562
     48 srwxr-xr-x    1 root     root             0 Jan 29 23:14 plugin771234044
     47 srwxr-xr-x    1 root     root             0 Jan 29 23:14 plugin894818988
 
time="2017-01-29T23:15:08Z" level=debug msg="agent: Received event" event="query: rpc:config" node=504404e2ade1 
time="2017-01-29T23:15:08Z" level=debug msg="agent: RPC Config requested" at=3 node=504404e2ade1 payload= query="rpc:config" 
time="2017-01-29T23:15:08Z" level=debug msg="proc: Received ack" from=504404e2ade1 node=504404e2ade1 query="rpc:config" 
time="2017-01-29T23:15:08Z" level=debug msg="proc: Received response" from=504404e2ade1 node=504404e2ade1 payload="172.17.0.4:6868" query="rpc:config" 
time="2017-01-29T23:15:11Z" level=debug msg="agent: Done receiving acks and responses" node=504404e2ade1 query="run:job" 
time="2017-01-29T23:15:11Z" level=debug msg="proc: Done receiving acks and responses" node=504404e2ade1 query="rpc:config" 
time="2017-01-29T23:15:11Z" level=debug msg="rpc: Received execution done" group=1485731708001177710 job=test2 node=504404e2ade1 
time="2017-01-29T23:15:11Z" level=debug msg="store: Retrieved job from datastore" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:11Z" level=debug msg="store: Setting key" execution=1485731708006166623-504404e2ade1 job=test2 node=504404e2ade1 
time="2017-01-29T23:15:11Z" level=debug msg="store: Retrieved job from datastore" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:11Z" level=debug msg="store: Setting job" job=test2 json="{\"name\":\"test2\",\"schedule\":\"@every 5s\",\"shell\":false,\"command\":\"ls -ali /tmp\",\"owner\":\"\",\"owner_email\":\"\",\"success_count\":1,\"error_count\":0,\"last_success\":\"2017-01-29T23:15:08.011256779Z\",\"last_error\":\"0001-01-01T00:00:00Z\",\"disabled\":false,\"tags\":null,\"retries\":0,\"dependent_jobs\":null,\"parent_job\":\"\",\"processors\":null,\"concurrency\":\"allow\"}" node=504404e2ade1 
time="2017-01-29T23:15:11Z" level=debug msg="rpc: from: %s504404e2ade1" node=504404e2ade1 
time="2017-01-29T23:15:13Z" level=debug msg="scheduler: Run job" job=test2 node=504404e2ade1 schedule="@every 5s" 
time="2017-01-29T23:15:13Z" level=debug msg="store: Retrieved job from datastore" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:13Z" level=debug msg="agent: Filtered nodes to run: []" node=504404e2ade1 
time="2017-01-29T23:15:13Z" level=debug msg="agent: Filtered tags to run: map[]" node=504404e2ade1 
time="2017-01-29T23:15:13Z" level=debug msg="agent: Sending query" job_name=test2 json="{\"execution\":{\"job_name\":\"test2\",\"started_at\":\"0001-01-01T00:00:00Z\",\"finished_at\":\"0001-01-01T00:00:00Z\",\"group\":1485731713002297607,\"attempt\":1},\"rpc_addr\":\"172.17.0.4:6868\"}" node=504404e2ade1 query="run:job" 
time="2017-01-29T23:15:13Z" level=debug msg="agent: Received event" event="query: run:job" node=504404e2ade1 
time="2017-01-29T23:15:13Z" level=debug msg="agent: Running job" at=4 node=504404e2ade1 payload="{\"execution\":{\"job_name\":\"test2\",\"started_at\":\"0001-01-01T00:00:00Z\",\"finished_at\":\"0001-01-01T00:00:00Z\",\"group\":1485731713002297607,\"attempt\":1},\"rpc_addr\":\"172.17.0.4:6868\"}" query="run:job" 
time="2017-01-29T23:15:13Z" level=info msg="agent: Starting job" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:13Z" level=debug msg="agent: Received ack" from=504404e2ade1 node=504404e2ade1 query="run:job" 
time="2017-01-29T23:15:13Z" level=debug msg="rpc: Received GetJob" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:13Z" level=debug msg="store: Retrieved job from datastore" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:13Z" level=debug msg="agent: GetJob by RPC" command="ls -ali /tmp" node=504404e2ade1 
time="2017-01-29T23:15:13Z" level=debug msg="agent: Received response" from=504404e2ade1 node=504404e2ade1 query="run:job" response="{\"job_name\":\"test2\",\"started_at\":\"2017-01-29T23:15:13.005675775Z\",\"finished_at\":\"0001-01-01T00:00:00Z\",\"node_name\":\"504404e2ade1\",\"group\":1485731713002297607,\"attempt\":1}" 
time="2017-01-29T23:15:13Z" level=debug msg="store: Setting key" execution=1485731713005675775-504404e2ade1 job=test2 node=504404e2ade1 
time="2017-01-29T23:15:13Z" level=debug msg="proc: Command output" node=504404e2ade1 output=total 8
     46 drwxrwxrwt    2 root     root          4096 Jan 29 23:14 .
      2 drwxr-xr-x   32 root     root          4096 Jan 29 23:14 ..
     49 srwxr-xr-x    1 root     root             0 Jan 29 23:14 plugin179447562
     48 srwxr-xr-x    1 root     root             0 Jan 29 23:14 plugin771234044
     47 srwxr-xr-x    1 root     root             0 Jan 29 23:14 plugin894818988

... 

```

##### Step 3

dkron logs while deleting from the UI:

```
time="2017-01-29T23:15:18Z" level=debug msg="agent: Received event" event="query: rpc:config" node=504404e2ade1 
time="2017-01-29T23:15:18Z" level=debug msg="agent: RPC Config requested" at=7 node=504404e2ade1 payload= query="rpc:config" 
time="2017-01-29T23:15:18Z" level=debug msg="proc: Received ack" from=504404e2ade1 node=504404e2ade1 query="rpc:config" 
time="2017-01-29T23:15:18Z" level=debug msg="proc: Received response" from=504404e2ade1 node=504404e2ade1 payload="172.17.0.4:6868" query="rpc:config" 
time="2017-01-29T23:15:18Z" level=debug msg="store: Retrieved leader from datastore" node=504404e2ade1 
time="2017-01-29T23:15:18Z" level=debug msg="store: Retrieved leader from datastore" node=504404e2ade1 
time="2017-01-29T23:15:21Z" level=debug msg="agent: Done receiving acks and responses" node=504404e2ade1 query="run:job" 
time="2017-01-29T23:15:21Z" level=debug msg="proc: Done receiving acks and responses" node=504404e2ade1 query="rpc:config" 
time="2017-01-29T23:15:21Z" level=debug msg="rpc: Received execution done" group=1485731718001655746 job=test2 node=504404e2ade1 
time="2017-01-29T23:15:21Z" level=debug msg="store: Retrieved job from datastore" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:21Z" level=debug msg="store: Setting key" execution=1485731718003194091-504404e2ade1 job=test2 node=504404e2ade1 
time="2017-01-29T23:15:21Z" level=debug msg="store: Retrieved job from datastore" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:21Z" level=debug msg="store: Setting job" job=test2 json="{\"name\":\"test2\",\"schedule\":\"@every 5s\",\"shell\":false,\"command\":\"ls -ali /tmp\",\"owner\":\"\",\"owner_email\":\"\",\"success_count\":3,\"error_count\":0,\"last_success\":\"2017-01-29T23:15:18.004290756Z\",\"last_error\":\"0001-01-01T00:00:00Z\",\"disabled\":false,\"tags\":null,\"retries\":0,\"dependent_jobs\":null,\"parent_job\":\"\",\"processors\":null,\"concurrency\":\"allow\"}" node=504404e2ade1 
time="2017-01-29T23:15:21Z" level=debug msg="rpc: from: %s504404e2ade1" node=504404e2ade1 
time="2017-01-29T23:15:22Z" level=debug msg="store: Retrieved job from datastore" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:22Z" level=debug msg="store: Retrieved leader from datastore" node=504404e2ade1 
time="2017-01-29T23:15:22Z" level=debug msg="agent: Received event" event="query: scheduler:restart" node=504404e2ade1 
time="2017-01-29T23:15:22Z" level=debug msg="agent: Restarting scheduler" node=504404e2ade1 
time="2017-01-29T23:15:22Z" level=debug msg="agent: Received ack" from=504404e2ade1 node=504404e2ade1 
time="2017-01-29T23:15:22Z" level=debug msg="scheduler: Stopping scheduler" node=504404e2ade1 
time="2017-01-29T23:15:25Z" level=debug msg="agent: Done receiving acks and responses" node=504404e2ade1 query="scheduler:restart" 
time="2017-01-29T23:15:28Z" level=debug msg="store: Retrieved leader from datastore" node=504404e2ade1 
time="2017-01-29T23:15:28Z" level=debug msg="store: Retrieved leader from datastore" node=504404e2ade1 
```

##### Step 4:

Re-creating job with the exact same POST

```
time="2017-01-29T23:15:52Z" level=debug msg="store: Setting job" job=test2 json="{\"name\":\"test2\",\"schedule\":\"@every 5s\",\"shell\":false,\"command\":\"ls -ali /tmp\",\"owner\":\"\",\"owner_email\":\"\",\"success_count\":0,\"error_count\":0,\"last_success\":\"0001-01-01T00:00:00Z\",\"last_error\":\"0001-01-01T00:00:00Z\",\"disabled\":false,\"tags\":null,\"retries\":0,\"dependent_jobs\":null,\"parent_job\":\"\",\"processors\":null,\"concurrency\":\"allow\"}" node=504404e2ade1 
time="2017-01-29T23:15:52Z" level=debug msg="store: Retrieved leader from datastore" node=504404e2ade1 
time="2017-01-29T23:15:52Z" level=debug msg="agent: Received event" event="query: scheduler:restart" node=504404e2ade1 
time="2017-01-29T23:15:52Z" level=debug msg="agent: Received ack" from=504404e2ade1 node=504404e2ade1 
time="2017-01-29T23:15:52Z" level=debug msg="agent: Restarting scheduler" node=504404e2ade1 
time="2017-01-29T23:15:52Z" level=debug msg="scheduler: Stopping scheduler" node=504404e2ade1 
time="2017-01-29T23:15:52Z" level=debug msg="scheduler: Adding job to cron" job=test2 node=504404e2ade1 
time="2017-01-29T23:15:55Z" level=debug msg="agent: Done receiving acks and responses" node=504404e2ade1 query="scheduler:restart" 
panic: runtime error: index out of range

goroutine 132 [running]:
panic(0x9fdb00, 0xc42000e090)
        /usr/local/opt/go/libexec/src/runtime/panic.go:500 +0x1a1
github.com/victorcoder/dkron/dkron.(*Store).GetLastExecutionGroup(0xc420047340, 0xc4207c7648, 0x5, 0x0, 0xc4202620a0, 0xc4200551e0, 0xc420638b38, 0x790350)
        /Users/victorcoder/src/github.com/victorcoder/dkron/dkron/store.go:279 +0x2ba
github.com/victorcoder/dkron/dkron.(*Job).Status(0xc420623b00, 0xc420638a01)
        /Users/victorcoder/src/github.com/victorcoder/dkron/dkron/job.go:128 +0x54
github.com/victorcoder/dkron/dkron.(*Job).isRunnable(0xc420623b00, 0xb1eb78)
        /Users/victorcoder/src/github.com/victorcoder/dkron/dkron/job.go:222 +0x40
github.com/victorcoder/dkron/dkron.(*Job).Run(0xc420623b00)
        /Users/victorcoder/src/github.com/victorcoder/dkron/dkron/job.go:100 +0xbe
created by github.com/victorcoder/dkron/cron.(*Cron).run
        /Users/victorcoder/src/github.com/victorcoder/dkron/cron/cron.go:159 +0x41d
```

**Every container restart leads to a similar crash**

```
/opt/local/dkron/dkron agent -server -backend zk -backend-machine 172.17.0.3:2181 -log-level=debug
2017/01/29 23:16:06 [DEBUG] plugin: starting plugin: /opt/local/dkron/dkron-processor-files []string{"/opt/local/dkron/dkron-processor-files"}
2017/01/29 23:16:06 [DEBUG] plugin: waiting for RPC address for: /opt/local/dkron/dkron-processor-files
2017/01/29 23:16:06 [DEBUG] plugin: dkron-processor-files: 2017/01/29 23:16:06 [DEBUG] plugin: plugin address: unix /tmp/plugin590231212
2017/01/29 23:16:06 [DEBUG] plugin: starting plugin: /opt/local/dkron/dkron-processor-log []string{"/opt/local/dkron/dkron-processor-log"}
2017/01/29 23:16:06 [DEBUG] plugin: waiting for RPC address for: /opt/local/dkron/dkron-processor-log
2017/01/29 23:16:06 [DEBUG] plugin: dkron-processor-log: 2017/01/29 23:16:06 [DEBUG] plugin: plugin address: unix /tmp/plugin665768699
2017/01/29 23:16:06 [DEBUG] plugin: starting plugin: /opt/local/dkron/dkron-processor-syslog []string{"/opt/local/dkron/dkron-processor-syslog"}
2017/01/29 23:16:06 [DEBUG] plugin: waiting for RPC address for: /opt/local/dkron/dkron-processor-syslog
2017/01/29 23:16:06 [DEBUG] plugin: dkron-processor-syslog: 2017/01/29 23:16:06 [DEBUG] plugin: plugin address: unix /tmp/plugin025283811
Starting Dkron agent...
time="2017-01-29T23:16:06Z" level=info msg="agent: Dkron agent starting" node=e293e9c37335 
time="2017-01-29T23:16:06Z" level=info msg="agent: joining: [] replay: true" node=e293e9c37335 
time="2017-01-29T23:16:06Z" level=debug msg="store: Backend config" backend=zk keyspace=dkron machines=[172.17.0.3:2181] node=e293e9c37335 
time="2017-01-29T23:16:06Z" level=info msg="api: Running HTTP server" address=":8080" node=e293e9c37335 
time="2017-01-29T23:16:06Z" level=debug msg="rpc: Registering RPC server" node=e293e9c37335 rpc_addr="172.17.0.4:6868" 
time="2017-01-29T23:16:06Z" level=info msg="agent: Running for election" node=e293e9c37335 
time="2017-01-29T23:16:06Z" level=info msg="agent: Listen for events" node=e293e9c37335 
time="2017-01-29T23:16:06Z" level=info msg="agent: Cluster leadership lost" node=e293e9c37335 
time="2017-01-29T23:16:07Z" level=debug msg="agent: Received event" event=member-join node=e293e9c37335 
time="2017-01-29T23:16:07Z" level=debug msg="agent: Member event" event=member-join member=e293e9c37335 node=e293e9c37335 
time="2017-01-29T23:16:08Z" level=info msg="agent: Cluster leadership acquired" node=e293e9c37335 
time="2017-01-29T23:16:08Z" level=debug msg="agent: Restarting scheduler" node=e293e9c37335 
time="2017-01-29T23:16:08Z" level=debug msg="scheduler: Adding job to cron" job=test2 node=e293e9c37335 
panic: runtime error: index out of range

goroutine 71 [running]:
panic(0x9fdb00, 0xc42000e090)
        /usr/local/opt/go/libexec/src/runtime/panic.go:500 +0x1a1
github.com/victorcoder/dkron/dkron.(*Store).GetLastExecutionGroup(0xc420046100, 0xc42012e2f8, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0)
        /Users/victorcoder/src/github.com/victorcoder/dkron/dkron/store.go:279 +0x2ba
github.com/victorcoder/dkron/dkron.(*Job).Status(0xc42013a240, 0x0)
        /Users/victorcoder/src/github.com/victorcoder/dkron/dkron/job.go:128 +0x54
github.com/victorcoder/dkron/dkron.(*Job).isRunnable(0xc42013a240, 0xb1eb78)
        /Users/victorcoder/src/github.com/victorcoder/dkron/dkron/job.go:222 +0x40
github.com/victorcoder/dkron/dkron.(*Job).Run(0xc42013a240)
        /Users/victorcoder/src/github.com/victorcoder/dkron/dkron/job.go:100 +0xbe
created by github.com/victorcoder/dkron/cron.(*Cron).run
        /Users/victorcoder/src/github.com/victorcoder/dkron/cron/cron.go:159 +0x41d
```

#### Work-around

The only way to make the container work again is to:

1. Start a `curl` requesting to delete job `test2`
2. Start the container

Adding the job back leads to the same error. Adding the same job with a different name works...

Am I missing something? Is there a specific zookeeper setup or version requirement? 

Cheers,

Andreas

**Comment by urban-1** [comment]
Update: Tried the exact same setup with 2x consul backend nodes instead of zk and it seems to be working as expected

**Comment by victorcoder** [comment]
Hi @urban-1 thanks for reporting, this should be caused by how zf treats the lifecycle of keys. I'll do som tests.

**Comment by leoleovich** [comment]
https://github.com/victorcoder/dkron/pull/252

**Comment by stephenranjit** [comment]
hi @victorcoder this is impeding testing when using same job name repetitively. dkron seems to be deleting the job correctly but still looks for the deleted keys (1test,2test) in store (ZK):
ERRO[2017-05-19T08:45:37-07:00] Key not found in store                        node=
INFO[2017-05-19T08:46:16-07:00] agent: Starting job                           job=1test node=
INFO[2017-05-19T08:47:16-07:00] agent: Starting job                           job=1test node=
INFO[2017-05-19T08:50:19-07:00] agent: Starting job                           job=2test node=

**Comment by walrusVision** [comment]
We're seeing the same error as @urban-1 in the same circumstances.

We're using dkron 0.9.3 against a 5 node zookeeper running 3.4.6.

**Comment by zwotzie** [comment]
we too, dkron 0.9.4 and 3 zookeeper nodes

**Comment by victorcoder** [comment]
I tried several times to reproduce with zk in standalone mode and it works as expected.

I'll do a test with > 1 nodes

**Comment by victorcoder** [comment]
I'm completely unable to reproduce it. I used a 3 nodes zk cluster to do the test:

1. Create a job `curl -X POST -d '{"name": "test2", "schedule": "@every 5s", "command": "ls -ali /tmp", "role": "web:1"}' localhost:32784/v1/jobs -vvv`
2. Remove the job via the UI
3. Recerate the job

Repeated this several times, no panic.

I tend to think that the newst version should fix the issue, release 0.9.5 is comming today, please, upgrade and try it again.

Closing it in the meantime.
